### ISSUE TITLE
Error message core_plugin/corrupted_archive_structure during installation of qtype_stack_moodle45_2024111500.zip

### ISSUE THREAD

**Issue by atrottmann** [maths/moodle-qtype_stack#1316]
When I try to install the current version qtype_stack_moodle45_2024111500.zip from https://moodle.org/plugins/qtype_stack/versions on different Moodles using the Moodle web interface, I get the error message:

```
    Debug-Info: Unknown error during zip extraction (file not created).
Error code: corrupted_archive_structure
$a contents: stack/samplequestions/stacklibrary/General-maths-questions/First-Order-ODEs/Differential_Equation_1.xml
```

The hosts in question are running Debian Bullseye and PHP Version 8.1.30 (including php-zip of the same version).

When I use `unzip` to unzip qtype_stack_moodle45_2024111500.zip manually and change the directory to stack/samplequestions/stacklibrary/General-maths-questions/ I see, among others, the following four directories:

```
drwxrwxrwx 2 ftc ftc 4096 18. Nov 13:33 "First-Order-ODE's"
drwxrwxrwx 2 ftc ftc 4096 18. Nov 13:33  First-Order-ODEs
drwxrwxrwx 2 ftc ftc 4096 18. Nov 13:33 "Second-Order-ODE'S"
drwxrwxrwx 2 ftc ftc 4096 18. Nov 13:33  Second-Order-ODES
```

The ones without the apostrophe are empty; the ones with the apostrophe contain XML files.

I think the problem involves the apostrophe in the directory name, but I don't know if all the files should be in the directories without apostrophe or the directories with apostrohe should not exist.

**Comment by aharjula** [comment]
Ahh, the joy of the eternal issues. I wonder what the file system and Zip version will be that will finally, without any problem, manage to encode stuff correctly at both ends. In this case that Zip from Moodle itself is already different from the content that comes from the Git. @sangwinc I guess this requires a hot-fix with simpler names for the directories; if you do that, maybe throw in a new JSXGraph as well; I should have remembered that before the release...

**Comment by sangwinc** [comment]
Thanks for reporting this.  I always install/update direct from github rather than downloading a zip, so I haven't encountered this problem.  We've moved files and directories in the samplequestions folder, and released v4.8.1.   I've downloaded the new zip files (windows and linux) and can't see any further problems.

**Comment by atrottmann** [comment]
Thank you very much I could now use the automatic update function of the Moodle plugins and it appears to have worked fine.
